test.disabled/ejabberd_tests/tests/mod_global_distrib_SUITE.erl

```diff
@@ -854,6 +887,58 @@ redis_query(Node, Query) ->
     RedisWorker = rpc(Node, wpool_pool, best_worker, [mod_global_distrib_mapping_redis]),
     rpc(Node, eredis, q, [RedisWorker, Query]).
 
+%% A fake address we don't try to connect to.
+%% Used in test_advertised_endpoints_override_endpoints testcase.
+advertised_endpoints() ->
+    [
+     {fake_domain(), 7777}
+    ].
+
+fake_domain() ->
+    "somefakedomain.com".
+
+iptuples_to_string([]) ->
+    [];
+iptuples_to_string([{Addr, Port} | Endps]) when is_tuple(Addr) ->
+    [{inet_parse:ntoa(Addr), Port} | iptuples_to_string(Endps)];
+iptuples_to_string([E | Endps]) ->
+    [E | iptuples_to_string(Endps)].
+
+maybe_add_advertised_endpoints(NodeName, Opts, Config) ->
+    Endpoints = proplists:get_value(NodeName, ?config(add_advertised_endpoints, Config), []),
+    case Endpoints of
+        [] ->
+            Opts;
+        E ->
+            Connections = case lists:keyfind(connections, 1, Opts) of
+                              false ->
+                                  [];
+                              C -> C
+                          end,
+            NewConnections = {connections, [{advertised_endpoints, E} | Connections]},
+            [NewConnections | Opts]
+    end.
+
+load_suite_module_on_nodes() ->
+    {_Module, Binary, Filename} = code:get_object_code(?MODULE),
+    execute_on_each_node(code, load_binary, [?MODULE, Filename, Binary]).
+
+mock_inet_on_each_node() ->
+    Nodes = lists:map(fun({NodeName, _, _}) -> ct:get_config(NodeName) end, get_hosts()),
+    lists:map(fun(Node) -> rpc:block_call(Node, ?MODULE, mock_inet, []) end, Nodes).
```

Comment:
escalus_ct:rpc_call/6 should probably used here. It would report an error.
We should check rpc call result at least.
Not sure, if we need a block call here, spawn should be enough.

Reply:
I have actually changed from "spawn-like" approach to "block_call-like" approach in result of a comment on this PR. So which one should be eventually used? @arcusfelis, @fenek?

Reply:
ok, I see reasoning behind block call.
Be aware, that if meck does not return in the call, we have a broken rpc calls.
Meck would also most likely link to the rpc process, which can cause issues. The maximum I can think of is crashing rpc process, that would be restarted. Can be fine with tests, but not OK for the server code ;)

So, you can keep it as it is for now.